### ISSUE TITLE
allow Boinc to use a variable number of cores

### ISSUE THREAD

**Issue by hpvpp** [BOINC/boinc#1590]
On multi-core cpu's allow the user to specify both the minimum and maximum number of cores available to Boinc.
The idea is to allow Boinc to reduce its cpu-load in a stepwise rather than all-or-nothing manner.

**Comment by ChristianBeer** [comment]
I don't understand when to use a stepwise reduction on cpu load? Do you want something that is described in #41?

**Comment by hpvpp** [comment]
rather, on multi-core systems:
while the value in  "suspend when non-Boinc cpu usage is above n %" is exceeded {
reduce the number of cpu's used by Boinc by one
}

**Comment by ChristianBeer** [comment]
This is what already happens because science apps have the lowest priority. What the suspend eliminates is disk IO, that the user may need for something else and the need for the OS scheduler to consider low priority processes when scheduling cpu runtime. So I'm not quite sure what the advantages of the stepwise reduction is compared to just setting a higher value and complete suspension. 

Asked another way: What's the overhead from having `L` low-priority and `N` normal priority processes for `C` cores running in parallel  with `L=C and N<=C` compared to reducing low-priority processes and have a ratio like `L+N=C and N<=C`. Normal priority processes means non-BOINC processes.

**Comment by hpvpp** [comment]
It just seemed to me a more intuitive way of doing things.
